src/core/thread/dua_manager.cpp

```diff
@@ -645,27 +649,41 @@ Error DuaManager::ProcessDuaResponse(Coap::Message &aMessage)
 #endif
 #if OPENTHREAD_FTD && OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
     {
-        Child *child = Get<ChildTable>().GetChildAtIndex(mChildIndexDuaRegistering);
+        Child *  child = nullptr;
+        uint16_t childIndex;
+
+        for (Child &iter : Get<ChildTable>().Iterate(Child::kInStateValidOrRestoring))
```

Comment:
Only iterate valid child here? Note that the parent only register for valid child.
https://github.com/openthread/openthread/blob/3a953086730eaad12f4319f5608e42bbc333727d/src/core/thread/dua_manager.cpp#L479-L488

Reply:
resolved.